### PR TITLE
DAOS-7174 pool: coverity fixes

### PR DESCRIPTION
**Pull request by kccain**
CID 331793 - unchecked return of uuid_parse()
CID 331792 - unchecked return of cmd_push_arg()

Signed-off-by: Kenneth Cain <kenneth.c.cain@intel.com>

### CODE REVIEW

File: src/engine/drpc_client.c [hunk 1]
```diff
@@ -390,7 +390,12 @@ ds_pool_find_bylabel(d_const_string_t label, uuid_t pool_uuid,
 	if (ranks == NULL)
 		D_GOTO(out_resp, rc = -DER_NOMEM);
 	*svc_ranks = ranks;
-	uuid_parse(frsp->uuid, pool_uuid);
+	rc = uuid_parse(frsp->uuid, pool_uuid);
+	if (rc != 0) {
+		D_ERROR("Unable to parse pool UUID %s: "DF_RC"\n", frsp->uuid,
+			DP_RC(rc));
+		D_GOTO(out_resp, rc = -DER_INVAL);
```
Comment (liw) [inline src/engine/drpc_client.c:397]: Generally, if a function like ds_pool_find_bylabel returns an error, its caller is not expected to look at pool_uuid and *svc_ranks. So we should free ranks before returning the -DER_INVAL. E.g., something like

    rc = uuid_parse(...);
    if (rc != 0) {
        ...
        D_FREE(*svc_ranks);
        D_GOTO(...);
    }

Additionally, is -DER_INVAL really appropriate? The caller hasn't specified any invalid argument in this case; it's the control plane supplying an invalid response...

**Review by liw — CHANGES_REQUESTED** [review]

File: src/engine/drpc_client.c (hunk 1 above)
Comment (liw) [reply src/engine/drpc_client.c:397]: Or better only assign ranks to *svc_ranks after frsp->uuid has been successfully parsed.

**Review by liw — APPROVED** [review]
Thanks.

File: src/engine/drpc_client.c [hunk 2]
```diff
@@ -385,12 +385,18 @@ ds_pool_find_bylabel(d_const_string_t label, uuid_t pool_uuid,
 		D_GOTO(out_resp, rc = frsp->status);
 	}
 
+	rc = uuid_parse(frsp->uuid, pool_uuid);
+	if (rc != 0) {
+		D_ERROR("Unable to parse pool UUID %s: "DF_RC"\n", frsp->uuid,
+			DP_RC(rc));
+		D_GOTO(out_resp, rc = -DER_IO);
```
Comment (mjmac) [inline src/engine/drpc_client.c:392]: minor nit: Is -DER_IO the right error here? Seems like maybe -DER_INVAL might be better?

**Review by mjmac — APPROVED** [review]
Generally LGTM, nothing to block on. Wondering about the error choice.

File: src/engine/drpc_client.c (hunk 2 above)
Comment (mjmac) [reply src/engine/drpc_client.c:392]: Oh, hah. I see the @liw requested a change from -DER_INVAL.

**Pull request merged**